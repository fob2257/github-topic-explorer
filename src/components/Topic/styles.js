import styled from 'styled-components';

export const Card = styled.div`
  box-shadow: rgb(0 0 0 / 15%) 0px 3px 6px 1px;
  margin: 16px 0px;
  padding: 16px;
`;

export const CardDiv = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const TopicName = styled.h2`
  margin: 0;
  color: #263238;
  font-weight: bold;
  font-size: 24px;
  letter-spacing: 0em;
`;

export const TopicID = styled.h3`
  margin: 0;
  color: #8a9094;
  font-weight: 400;
  font-size: 14px;
  line-height: 2;
  letter-spacing: 0.00938em;
`;
