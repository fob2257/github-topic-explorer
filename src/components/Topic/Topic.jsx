import React from 'react';

import { Card, CardDiv, TopicName, TopicID } from './styles';
import RelatedTopics from '../RelatedTopics/RelatedTopics';
import StargazerCount from '../StargazerCount/StargazerCount';

const Topic = ({ topic }) => (
  <Card>
    <CardDiv>
      <div>
        <TopicName>{topic.name}</TopicName>
        <TopicID>{topic.id}</TopicID>
      </div>

      <StargazerCount count={topic.stargazerCount} />
    </CardDiv>

    <RelatedTopics topics={topic.relatedTopics} />
  </Card>
);

export default Topic;
