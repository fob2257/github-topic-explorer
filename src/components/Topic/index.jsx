import React from 'react';

import { Card, CardDiv, TopicName, TopicID } from './styles';
import RelatedTopics from '../RelatedTopics';
import StargazerCount from '../StargazerCount';

const Topic = ({ topic }) => (
  <Card>
    <CardDiv>
      <div>
        <TopicName>{topic.name}</TopicName>
        <TopicID>{topic.id}</TopicID>
      </div>

      <StargazerCount count={topic.stargazerCount} />
    </CardDiv>

    <RelatedTopics topics={topic.relatedTopics} />
  </Card>
);

export default Topic;
