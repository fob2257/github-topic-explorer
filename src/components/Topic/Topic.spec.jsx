import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import Topic from './Topic';

const topic = {
  __typename: 'Topic',
  id: 'MDU6VG9waWNhbmd1bGFy',
  name: 'angular',
  stargazerCount: 39188,
  relatedTopics: [
    {
      __typename: 'Topic',
      id: 'MDU6VG9waWN2dWU=',
      name: 'vue',
      stargazerCount: 45009,
    },
  ],
};

describe('<Topic />', () => {
  const setup = () => <Topic topic={topic} />;

  it('should render correctly', () => {
    render(setup());

    const element = screen.getByRole('heading', { name: new RegExp(topic.name, 'i') });

    expect(element).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const component = create(setup());
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
