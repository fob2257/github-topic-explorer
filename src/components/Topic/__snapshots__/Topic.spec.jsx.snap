// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topic /> should match snapshot 1`] = `
<div
  className="sc-bdvvtL eDSkTd"
>
  <div
    className="sc-gsDKAQ kLyXFC"
  >
    <div>
      <h2
        className="sc-dkPtRN bwmxL"
      >
        angular
      </h2>
      <h3
        className="sc-hKwDye cmAQBw"
      >
        MDU6VG9waWNhbmd1bGFy
      </h3>
    </div>
    <div
      className="sc-jrQzAO gsHAeA"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="StarIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <span
        className="sc-kDTinF cicfqr"
      >
        39188
      </span>
    </div>
  </div>
  <div
    className="sc-eCImPb iZszzz"
  >
    <h2
      className="sc-jRQBWg cAiQqT"
    >
      Related topics: 
      1
    </h2>
    <div
      className="sc-gKclnd jndObI"
      onClick={[Function]}
    >
      <h3
        className="sc-iCfMLu cuhnuA"
      >
        vue
      </h3>
      <div
        className="sc-jrQzAO gsHAeA"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="StarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <span
          className="sc-kDTinF cicfqr"
        >
          45009
        </span>
      </div>
    </div>
  </div>
</div>
`;
